app/src/main/java/com/m/services/dataSync/TaskActivity.java

```diff
@@ -106,4 +105,33 @@ private void updateUi(){
         runOnUiThread(() -> itemAdapter.notifyDataSetChanged());
     }
 
+    private void uiToast(String message) {
+        runOnUiThread(() -> Toast.makeText(getApplicationContext(), message, Toast.LENGTH_LONG).show());
+
+    }
+
+    private void handleErrorExceptions(ApolloException error){
+        if (error instanceof ApolloHttpException){
+            handleHttpErrors((ApolloHttpException) error);
+        } else if (error instanceof ApolloNetworkException){
+            uiToast(error.getMessage());
+        } else {
+            uiToast(error.getMessage());
+        }
+    }
+
+    private void handleHttpErrors(ApolloHttpException error){
+      switch (error.code()){
+          case 401:
+          case 403:
+              loginController.reAuthorise();
+              break;
+          case 503:
```

Comment:
just an idea, I have spoken with Jim about it and we are not 100pc sure if that's a good idea but would it make any sense to use cache if there is a server error? I think it would be nice if the app still works if server breaks down or takes too much time to respond.

Reply:
It is not in the requirement, but if it's easy to do then certainly we can try. But user should be notified that it is cache data being returned if this happens.